src/cplscheme/SerialCouplingScheme.hpp

```diff
@@ -76,9 +76,14 @@ class SerialCouplingScheme : public BiCouplingScheme {
    * @brief SerialCouplingSchemes applies acceleration to send data
    * @returns DataMap being accelerated
    */
-  DataMap &getAccelerationData() override
+  DataMap getAccelerationData() override
   {
-    return getSendData();
+    // create a copy of send data for read-only access.
+    DataMap accelerationData;
+    for (auto &dataPair : getSendData()) {
+      accelerationData.emplace(dataPair);
+    }
+    return accelerationData;
```

Comment:
Can't you simply return the send data?

```cpp
return getSendData();
```

Reply:
Yes. I think this was introduced by me on the fly and now it's not needed anymore. Removed it again in 701d134fbe1b53007856d615f85b48dcd0130c74.